sudoku.go

```diff
@@ -0,0 +1,103 @@
+package sudoku
+
+import "fmt"
+
+// Solve is the main task of this app
+func Solve(puzzle []int) ([]int, error) {
+	if len(puzzle) != 81 {
+		return nil, fmt.Errorf("the sudoku has to have 9 rows and 9 columns")
+	}
+	solved := backtrack(puzzle)
+	if !solved {
+		return nil, fmt.Errorf("the sudoku was impossible to solve")
+	}
+	return puzzle, nil
+}
+
+func backtrack(puzzle []int) bool {
+	if isSolved(puzzle) {
```

Comment:
I like the fact you want to check if puzzle is already correct and the tests cover this path.

a way to expand on this would be to have two go routines with ctx and cancel the longer algo job if the puzzle was solved.

this would show you can use ctx and go routines